### PR TITLE
Idaho partial deductions

### LINKED ISSUE #2921: Idaho household and dependent care deduction
[https://law.justia.com/codes/idaho/2022/title-63/chapter-30/section-63-3022d/](url)

[chrome-extension://efaidnbmnnnibpcajpcglclefindmkaj/https://tax.idaho.gov/wp-content/uploads/forms/EIN00046/EIN00046_03-01-2023.pdf](url)

### PR DESCRIPTION
**Pull request by jialyu0411**
Fixes #2921

Co-authored-by: Pavel Makarchuk <PavelMakarchuk@users.noreply.github.com>

### Summary
📝📄🧮

This pull request implements the household and dependent care services deductions for Idaho income tax in the policy engine. It adds a new parameter file `deductions.yaml` and a new variable class `IDDeductions` under the `id` state module.

> _`IdahoDeductions`_
> _New variable and file_
> _Autumn leaves fall fast_

### Walkthrough
* Bump minor version and add changelog entry for Idaho deductions ([link](https://github.com/PolicyEngine/policyengine-us/pull/2935/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1-R4))
* Create parameter file for Idaho deductions with list, description, and reference ([link](https://github.com/PolicyEngine/policyengine-us/pull/2935/files?diff=unified&w=0#diff-b8356111c77589702dd5a8cfb5c5321f09573ab04c26bc9f0560d944fd593f7aR1-R9))
* Define variable class for Idaho deductions with value type, entity, label, unit, definition period, and state code ([link](https://github.com/PolicyEngine/policyengine-us/pull/2935/files?diff=unified&w=0#diff-8a2cb07e34c3dbafe15fd2db2184bd272a6be844efbd403fbcf4caf7d78830ffR1-R11))

### CODE REVIEW

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/deductions.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+description: Idaho accounts for these deductions of the agjusted gross income.
+values:
+  2021-01-01:
+    - cdcc
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/deductions.yaml:4]: The source says it's the expenses, not the federal credit

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/deductions.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+description: Idaho accounts for these deductions of the agjusted gross income.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/deductions.yaml:1]: Though is this true? Hard to tell from this reference. Is it the list of itemized deductions? Does Idaho describe subtractions from AGI as "deductions"? (Many states call them subtractions, where deductions factor into taxable income instead). Does Idaho have a standard deduction?
```suggestion
description: Idaho deducts these items when calculating state adjusted gross income.
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Comment by PavelMakarchuk** [comment]
this PR adds the subtractions which do not have a separate calculation. Other subtractions are calculated in separate PR's and added to the list

File: policyengine_us/parameters/gov/states/id/tax/income/subtractions/subtractions.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+description: Idaho subtracts these items when calculating state adjusted gross income.
+values:
+  2021-01-01:
+    - cdcc_relevant_expenses
+    - us_govt_interest
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/id/tax/income/subtractions/subtractions.yaml:5]: Similarly reference subsection for other items.
```suggestion
    - us_govt_interest # (3)(a)
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
Please also hook up id_subtractions to the net income tree

File: policyengine_us/parameters/gov/states/id/tax/income/subtractions/subtractions.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+description: Idaho subtracts these items when calculating state adjusted gross income.
+values:
+  2021-01-01:
+    - cdcc_relevant_expenses
```
[the remark this answers is not in the document]
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/id/tax/income/subtractions/subtractions.yaml]: we will make the ID cdcc in a separate PR -> removed

File: policyengine_us/variables/gov/states/id/tax/income/id_taxable_income.py [hunk 5]
```diff
@@ -8,3 +8,6 @@ class id_taxable_income(Variable):
     unit = USD
     definition_period = YEAR
     defined_for = StateCode.ID
+
+    adds = ["id_additions", "adjustmented_gross_income"]
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/id/tax/income/id_taxable_income.py:12]:
```suggestion
    adds = ["id_additions", "adjusted_gross_income"]
```

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**